template/template.go

```diff
@@ -58,8 +59,11 @@ func New(options ...Option) (*Template, error) {
 		o(t.text, t.html)
 	}
 
-	t.text.Funcs(tmpltext.FuncMap(DefaultFuncs))
-	t.html.Funcs(tmplhtml.FuncMap(DefaultFuncs))
+	sprigFuncMap := sprig.FuncMap()
+	delete(sprigFuncMap, "env")
```

Comment:
I had another thought about how we can make this even safer. Instead of deleting functions from the map, let's declare our own map of Sprig functions that we know are safe.

The benefit here is if new, un-safe functions are added to Sprig, then we cannot introduce them accidentally by just updating the package in go.mod. Instead, we will need to add them to the allowlist of functions.

What do you think?

Reply:
I think having some kind of safe func map will be critical to merging this PR. I don't think it matters whether that func map is in Alertmanager or in Sprig though.

The reason is that Alertmanager is also used in multi-tenanted systems such as Mimir, in Cloud Managed environments. Bad actors could use Alertmanager templates to attempt to read secrets from environment variables, or traverse internal services using `getHostByName`.

Reply:
@grobinson-grafana Thanks for this insightful feedback. I was not aware of thoses aspects. I will rethink this and rework the MR. 
Also, in the meantime, I had a look at the sprig project. The chances of getting a change like this merged there are slim to none.
So either I will still include sprig but just select dedicated functions or I consider duplicating stuff. There is `HermeticHtmlFuncMap()` which looks promising.

Reply:
Changed to only allow functions explicitly listed.